Vizzuality/heco-invest
# feat: Notify PD/Investors on account deletion

This PR adds all required notifications when:
 - open call is deleted --> notifies investor and project developers which applied for it
 - project is deleted --> notifies involved project developers and investor which had application for this project
 - account is deleted --> all use cases described at https://vizzuality.atlassian.net/browse/LET-876

I was playing with idea if it wouldn't be easier to move email sending to callbacks, but have bad experience with it so decided in the end to keep it in controllers. To keep controllers slim, I have moved all notification logic to new services which are responsible for sending emails upon deletion.

## Testing instructions
  - Apply to some Open call as project developer and delete this open call as investor. Project developer should be notified.
  - Apply to some Open call as project developer and delete project which you have applied with. Investor of open call should be notified.
  - Apply to some Open call as project developer and delete project developer --> project developer should be notified and investor of open call which you have applied for.
  - Apply to some Open call as project developer and delete investor of open call --> investor should be notified and project developer of project that application was for.

All use cases should be testable via backoffice, first two are also testable via rswag documentation.

## Tracking

https://vizzuality.atlassian.net/browse/LET-915
https://vizzuality.atlassian.net/browse/LET-909
https://vizzuality.atlassian.net/browse/LET-877
https://vizzuality.atlassian.net/browse/LET-881
